tests/integration/src/test/scala/mesosphere/marathon/integration/UpgradeIntegrationTest.scala

```diff
@@ -215,8 +215,10 @@ class UpgradeIntegrationTest extends AkkaIntegrationTest with MesosClusterTest w
       (marathonCurrent.client.info.entityJson \ "version").as[String] should be(BuildInfo.version.toString)
 
       Then("All apps from 1.4.9 and 1.5.15 are still running")
-      marathonCurrent.client.tasks(app_149.id.toPath).value should contain theSameElementsAs (originalApp149Tasks)
-      marathonCurrent.client.tasks(app_1515.id.toPath).value should contain theSameElementsAs (originalApp1515Tasks)
+      val originalApp149TasksWithRole = originalApp149Tasks.map(t => t.copy(role = Some("foo")))
+      val originalApp1515TasksWithRole = originalApp1515Tasks.map(t => t.copy(role = Some("foo")))
+      marathonCurrent.client.tasks(app_149.id.toPath).value should contain theSameElementsAs (originalApp149TasksWithRole)
```

Comment:
Hm, why not just compare `marathonCurrent.client.tasks(app_149.id.toPath).value.map(_.id) should contain theSameElementsAs (originalApp149TaskIds)`?

Reply:
Yeah, comparing IDs is better.